client/src/main/java/org/eclipse/hono/client/impl/AbstractHonoClient.java

```diff
@@ -165,8 +165,16 @@ protected static final void setApplicationProperties(final Message msg, final Ma
             sender.setQoS(qos);
             sender.openHandler(senderOpen -> {
                 if (senderOpen.succeeded()) {
-                    LOG.debug("sender open [{}]", sender.getRemoteTarget());
-                    result.complete(senderOpen.result());
+                    LOG.info("sender open [{}] sendQueueFull [{}]", sender.getRemoteTarget(), sender.sendQueueFull());
+                    // wait on credits a little time, if not already given TODO: configurable waiting time
+                    if (sender.sendQueueFull()) {
+                        ctx.owner().setTimer(10, timerID -> {
+                            LOG.info("waited 10ms on credits [{}]", sender.getCredit());
+                            result.complete(senderOpen.result());
```

Comment:
this should probably also be `sender` instead of `senderOpen.result()`, just to be consistent with the rest of the handler's code?